### PR TITLE
Dict cursor support

### PR DESCRIPTION
**Pull request by xch91**
For those migrated from a data platform that uses dict cursor to iterate data set :)

### CODE REVIEW

File: impala/tests/test_hive_dict_cursor.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+def test_dict_cursor(cur):
+    cur.execute('CREATE TABLE tmp_hive (a STRING, b INT, c DOUBLE)')
+
+    cur.execute('SHOW TABLES')
+    tables = cur.fetchall()
+    assert any([t[0] == 'tmp_hive' for t in tables])
```
Comment (julienbaley) [inline impala/tests/test_hive_dict_cursor.py]: Square brackets superfluous here, no need to create a list.

File: impala/tests/test_hive_dict_cursor.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    cur.execute("SELECT c from tmp_hive WHERE a = 'bar'")
+    results = cur.fetchall()
+    assert len(results) == 1
+    assert results[0]['c'] is None
+
+    cur.execute('DROP TABLE tmp_hive')
+
+    cur.execute('SHOW TABLES')
+    tables = cur.fetchall()
+    assert all([t['tableName'] != 'tmp_hive' for t in tables])
```
Comment (julienbaley) [inline impala/tests/test_hive_dict_cursor.py]: Same here for the square brackets. 
Also, I think 'not any =='  reads better (as English!) than 'all  !='

**Comment by xch91** [comment]
Hi @julienbaley , are we merging this PR? Else I need to maintain a local version to support dict cursor in legacy code XD

**Comment by julienbaley** [comment]
I'm not part of the maintaining team! :) LGTM otherwise.

**Comment by xch91** [comment]
LOL! Let's see how maintaining team decides then. Thanks a looooot pal

**Comment by wesm** [comment]
Looks good here. Merging, thank you

**Comment by haosdent** [comment]
Orz

**Pull request merged**